pkg/metadata/prometheus.go

```diff
@@ -31,6 +33,13 @@ func (p *Prometheus) MetricMetadata(r *metadatapb.MetricMetadataRequest, s metad
 		return err
 	}
 
-	return s.Send(&metadatapb.MetricMetadataResponse{Result: &metadatapb.MetricMetadataResponse_Metadata{
-		Metadata: metadatapb.FromMetadataMap(md)}})
+	tracing.DoInSpan(s.Context(), "send_metadata_response", func(_ context.Context) {
+		err = s.Send(&metadatapb.MetricMetadataResponse{Result: &metadatapb.MetricMetadataResponse_Metadata{
+			Metadata: metadatapb.FromMetadataMap(md)}})
+	})
+	if err != nil {
+		return err
+	}
```

Comment:
I think you misinterpreted my previous suggestion, you should remove this block.
```suggestion
```

Reply:
@KrishnaSindhur the issue hasn't resolved so marking it unresolved. You don't need to check if err is nil.

Reply:
ya updated